src/cdk/stepper/stepper.ts

```diff
@@ -124,9 +124,12 @@ export class CdkStep implements OnChanges {
   /** The top level abstract control of the step. */
   @Input() stepControl: AbstractControlLike;
 
-  /** Whether user has seen the expanded step content or not. */
+  /** Whether user has attempted to move away from the step. */
   interacted = false;
 
+  /** Emits when the user has attempted to move away from the step. */
+  @Output('interacted') interactedStream: EventEmitter<CdkStep> = new EventEmitter<CdkStep>();
```

Comment:
I'm a little on the fence about the naming here. I think that `interacted` sounds more like an event name, but we already used it for the `interacted` property. I'm open to other naming suggestions.

Reply:
Would it make sense to call it `touched` or is this a slightly different idea?

Reply:
It sort of is like `touched`, but I was trying to avoid using the same terminology is forms to avoid confusion.

Reply:
`interacted` sounds fine to me, but I think we normally call our streams `xChanges`

cc @jelbourn in case he has any naming opinions